src/detect-engine-uint.c

```diff
@@ -88,6 +88,16 @@ static int DetectU32Validate(DetectU32Data *du32)
                 return 1;
             }
             break;
+        case DETECT_UINT_LTE:
+            if (du32->arg1 == UINT32_MAX) {
```

Comment:
Having a really hard time understanding these validations. I know it is not wrong since it's already been there for so long but how does checking for the first arg to be `UINT32_MAX` in case of `LTE` help us validate the argument? Also, why do we not check anything w.r.t. `arg2`?

Reply:
`DETECT_UINT_LTE` and `UINT32_MAX` comes from `<=0xffffffff` which is always true, so we do not need a rule

Arg2 is only used with ranges